cypress/e2e/identify.cy.js

```diff
@@ -12,4 +12,56 @@ describe('identify()', () => {
         cy.posthog().invoke('identify', 'some-id')
         cy.posthog().invoke('opt_out_capturing')
     })
+
+    it('merges people as expected when reset is called', () => {
+        cy.posthog().invoke('capture', 'an-anonymous-event')
+        cy.posthog().invoke('identify', 'first-identify') // test identify merges with previous events after init
+        cy.posthog().invoke('capture', 'an-identified-event')
+        cy.posthog().invoke('identify', 'second-identify-should-not-be-merged') // test identify is not sent after previous identify
+        cy.posthog().invoke('capture', 'another-identified-event') // but does change the distinct id
+        cy.posthog().invoke('reset')
+        cy.posthog().invoke('capture', 'an-anonymous-event')
+        cy.posthog().invoke('identify', 'third-identify')
+        cy.posthog().invoke('capture', 'an-identified-event')
+
+        cy.phCaptures({ full: true }).then((events) => {
+            const eventsSeen = events.map((e) => e.event)
+
+            expect(eventsSeen.filter((e) => e === '$identify').length).to.eq(2)
+
+            expect(eventsSeen).to.deep.eq([
+                '$pageview',
+                'an-anonymous-event',
+                '$identify',
+                'an-identified-event',
+                'another-identified-event',
+                'an-anonymous-event',
+                '$identify',
+                'an-identified-event',
+            ])
+
+            expect(new Set(events.map((e) => e.properties['$device_id'])).size).to.eql(1)
+
+            // the first two events share a distinct id
```

Comment:
this is a pretty verbose test but I want to prove the chain of behaviour not just the steps in isolation